File: ExamenUnidad2__JAVA/ExamenUnidad2/src/examenunidad2/Mantenimiento.java

package examenunidad2;

public class Mantenimiento
{
    
   private static int IDENTIFICADOR = 200;
    private int id;
    private int idEncargado;
    private String tipoDeMantenimiento;
    private String fechaProceso;
    private String observaciones;
    
    public Mantenimiento(int idEncargado,String tipoDeMantenimiento, String fechaProceso, String observaciones ) 
    {
        this.idEncargado=idEncargado;
        this.tipoDeMantenimiento = tipoDeMantenimiento;
        this.fechaProceso = fechaProceso;
        this.observaciones = observaciones;
        this.id=IDENTIFICADOR;
        IDENTIFICADOR++;
    }
 
    public Mantenimiento(int idEncargado, String tipoDeMantenimiento, String fechaProceso)
    {
        this.idEncargado = idEncargado;
        this.tipoDeMantenimiento = tipoDeMantenimiento;
        this.fechaProceso = fechaProceso;
        this.id=IDENTIFICADOR;
        IDENTIFICADOR++;
    }
    
    
    public int getId() {
        return id;
    }
 
    
    public int getIdEncargado() {
        return idEncargado;
    }
    
    
    public String getInformacionMantenimiento()
    {
        return String.format("\nId: %d \nTipo de mantenimiento: %s \nFecha de mantenimiento: %s \nGuia encargado: %d \nObservaciones: %s",
            id,tipoDeMantenimiento, fechaProceso, idEncargado, observaciones );
    }
 
    

}

File: ExamenUnidad2_PYTHON 1/ExamenUnidad2_PYTHON 1/ExamenUnidad2/Mantenimiento.py
class Mantenimiento:
    IDENTIFICADOR = 200

    def __init__(self, id_encargado, tipo_de_mantenimiento, fecha_proceso, observaciones=None):
        self.id_encargado = id_encargado
        self.tipo_de_mantenimiento = tipo_de_mantenimiento
        self.fecha_proceso = fecha_proceso
        self.observaciones = observaciones
        self.id = Mantenimiento.IDENTIFICADOR
        Mantenimiento.IDENTIFICADOR += 1

        
    def get_id(self):
        return self.id

    def get_id_encargado(self):
        return self.id_encargado
    
    def get_informacion_mantenimiento(self):
        return f"\nId: {self.id} \nTipo de mantenimiento: {self.tipo_de_mantenimiento} \nFecha de mantenimiento: {self.fecha_proceso} \nGuía encargado: {self.id_encargado} \nObservaciones: {self.observaciones}" if self.observaciones else f"\nId: {self.id} \nTipo de mantenimiento: {self.tipo_de_mantenimiento} \nFecha de mantenimiento: {self.fecha_proceso} \nGuía encargado: {self.id_encargado}"
